Analysis/src/EventTags.cpp

```diff
@@ -73,31 +73,39 @@ EventTags EventTagCreator::CreateEventTags(const DataId& dataId_base, float weig
     };
 
     EventTags evt_tags;
-
     if(!event_unc_sources.count(dataId_base.Get<UncertaintySource>()))
         return evt_tags;
 
+
     for(const auto& category : categories) {
         if(!category.Contains(static_cast<size_t>(num_central_jets), bjet_counts, is_vbf, is_boosted, vbf_cat.second))
             continue;
         EventSubCategory evtSubCategory;
         const float btag_sf = category.HasBtagConstraint() && !is_data ? get_weight_btag(category.BtagWP()) : 1.f;
         const float cat_weight = weight * btag_sf;
 
-
         if(has_b_pair) {
             const EllipseParameters& window = category.HasBoostConstraint() && category.IsBoosted()
                     ? cuts::hh_bbtautau_Run2::hh_tag::boosted_window : cuts::hh_bbtautau_Run2::hh_tag::resolved_window;
             evtSubCategory.SetCutResult(SelectionCut::mh, SVfit_valid && window.IsInside(SVfit_p4.M(), m_bb));
             evtSubCategory.SetCutResult(SelectionCut::KinematicFitConverged, kinFit_convergence > 0);
         }
-
         for(const auto& subCategory : subCategories) {
             if(!evtSubCategory.Implies(subCategory)) continue;
-            const auto& dataId = dataId_base.Set(category).Set(subCategory);
+            const auto dataId = dataId_base.Set(category).Set(subCategory);
             evt_tags.dataIds.push_back(dataId);
             evt_tags.weights.push_back(cat_weight);
+            if(!is_data && dataId_base.Get<UncertaintySource>()==UncertaintySource::None) {
+                for(const auto& [key, weight_shift] : unc_map) {
+                    if(!category.HasBtagConstraint() && analysis::ToString(key.first).find("btag")!=std::string::npos) continue;
```

Comment:
don't perform string search - it is very slow. Have a predefined set of btag unc